streams/src/main/java/org/apache/kafka/streams/processor/internals/StateDirectory.java

```diff
@@ -215,14 +216,18 @@ synchronized void unlockGlobalState() throws IOException {
     synchronized void unlock(final TaskId taskId) throws IOException {
         final LockAndOwner lockAndOwner = locks.get(taskId);
         if (lockAndOwner != null && lockAndOwner.owningThread.equals(Thread.currentThread().getName())) {
-            locks.remove(taskId);
-            lockAndOwner.lock.release();
+            try {
+                lockAndOwner.lock.release();
+            } catch (final ClosedChannelException cce) {
```

Comment:
Should we catch `IOException` here? Or to be more precise: have a `catch` block for both, and if `IOException is thrown, rethrow it at the end instead of a clean return? Or do we expect that we cannot close the `FileChannel` anyway if we hit `IOException` here?